tar/test/test_crlf_mtree.c

```diff
@@ -0,0 +1,71 @@
+/*-
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright (c) 2018 The FreeBSD Foundation
+ * All rights reserved.
+ *
+ * This software was developed by Arshan Khanifar <arshankhanifar@gmail.com>
+ * under sponsorship from the FreeBSD Foundation.
+ */
+#include "test.h"
+
+DEFINE_TEST(test_crlf_mtree)
+{
+	char *p0;
+	size_t s;
+	int r;
+	p0 = NULL;
+	char *content = "#mtree\r\nf type=file uname=\\\r\nroot gname=root mode=0755 content=bar/foo\r\ng type=file uname=root gname=root mode=0755 content=bar/goo\r\n";
```

Comment:
Nit:  This would be easier to read if you reformatted it a bit, taking advantage of Standard C string concatenation:
```
char *content = "#mtree\r\n"
    "f type=file uname=\\\r\n"
    "root gname=root mode=0755 content=bar/foo\r\n"
    "g type=file uname=root gname=root mode=0755 content=bar/goo\r\n";
```

Note how this makes the individual lines of input much clearer.